addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerJsonParser.java

```diff
@@ -280,6 +281,74 @@ private static ImageIcon createSvgIcon(URL url) throws Exception {
+                            try {
+                                Map<String, Map<String, Map<String, AppPattern>>> domSelectorMap =
+                                        new HashMap<>();
+                                Map<String, Map<String, AppPattern>> nodeSelectorMap =
+                                        new HashMap<String, Map<String, AppPattern>>();
+                                Map<String, AppPattern> value = new HashMap<String, AppPattern>();
+                                appPat = this.strToAppPattern(type, (String) ent.getValue());
+                                value.put((String) ent.getKey(), appPat);
+                                nodeSelectorMap.put((String) entr.getKey(), value);
+                                domSelectorMap.put((String) (entry).getKey(), nodeSelectorMap);
+                                list.add(domSelectorMap);
+                            } catch (NumberFormatException e) {
+                                logger.error(
+                                        "Invalid field syntax "
+                                                + ent.getKey()
+                                                + " : "
+                                                + ent.getValue(),
+                                        e);
+                            } catch (PatternSyntaxException e) {
+                                patternErrorHandler.handleError((String) ent.getValue(), e);
+                            }
+                        }
+                    } else {
+                        try {
+                            Map<String, Map<String, Map<String, AppPattern>>> domSelectorMap =
+                                    new HashMap<>();
+                            Map<String, Map<String, AppPattern>> nodeSelectorMap = new HashMap<>();
+                            Map<String, AppPattern> value = new HashMap<String, AppPattern>();
+                            appPat = this.strToAppPattern(type, (String) entr.getValue());
+                            value.put((String) entr.getKey(), appPat);
+                            nodeSelectorMap.put((String) entr.getKey(), value);
+                            domSelectorMap.put((String) (entry).getKey(), nodeSelectorMap);
+                            list.add(domSelectorMap);
+                        } catch (NumberFormatException e) {
+                            logger.error(
+                                    "Invalid field syntax "
+                                            + entr.getKey()
+                                            + " : "
+                                            + entr.getValue(),
+                                    e);
+                        } catch (PatternSyntaxException e) {
+                            patternErrorHandler.handleError((String) entr.getValue(), e);
+                        }
+                    }
+                }
+            }
+        } else if (json != null) {
```

Comment:
This check could be done earlier and return (`Collections.emptyList()`) to avoid nesting.